### PR TITLE
use simplfied Bitmap

### PR DESCRIPTION
**Pull request by demoray**
Replace `bitmaps::Bitmap` with a simplified bitmap implementation for the 256 wide use case needed by boreal.

This does a few things:
* Provide a simplified interface for Bitmap based on what's used by boreal
* Replaces a bunch of casts & slice indexing that could cause future issues with compile-time validation (by moving from indexing into the Bitmap by usize to u8)
* Removes a MPL2 licensed dependency (which can be interpreted as incompatible with MIT/Apache)

### CODE REVIEW

**Comment by demoray** [comment]
note: the coverage task is currently failing due to issues addressed in #122.

**Comment by vthib** [comment]
Sorry about all the CI issues. I wouldn't mind removing a dependency, especially as we only use the 256 sized bitmap, so making a simple, custom impl makes sense. I'll try to review this before the end of the week

File: boreal/src/bitmaps.rs
Comment (vthib) [inline boreal/src/bitmaps.rs:85]: This function can be a bit simpler:

```rust
let (v, offset) = if self.bitmap.low != 0 {
    (&mut self.bitmap.low, 0)
} else if self.bitmap.high != 0 {
    (&mut self.bitmap.high, 128)
} else {
    return None;
};

// Safety: this value is contained in [0; 127] so it always fits in a u8.
let t: u8 = v.trailing_zeros().try_into().unwrap();
*v &= !(1 << t);
Some(offset + t)
```

the generated assembly is reduced as well

File: boreal/src/matcher/literals.rs
Comment (vthib) [inline boreal/src/matcher/literals.rs:201]: This allow can be removed i believe?

File: boreal/src/matcher/literals.rs
Comment (vthib) [inline boreal/src/matcher/literals.rs:300]: idem

File: boreal/src/bitmaps.rs
Comment (vthib) [inline boreal/src/bitmaps.rs:46]: This is actually only called with `value=true`, except in the iterator implementation. but with my iter implementation suggestion, it is actually no longer used at all, so it can be simplified and the value removed: there is no api to set a bit to false (this is fine for our use of the bitmap)

**Review by vthib — CHANGES_REQUESTED** [review]
Thanks for this contribution, I have a few comments but otherwise lgtm. I'll probably add a few additional tests afterwards but it's fine for this PR

**Comment by demoray** [comment]
@vthib I've applied your suggestions.

**Review by vthib — APPROVED** [review]

**Comment by vthib** [comment]
@demoray approved, but needs a rebase, probably after the merges of the nightly fixes.

**Pull request merged**
